.travis.yml

```diff
@@ -115,48 +114,109 @@ install:
-#   - cd ./contrib; git clone https://github.com/nanodbc/nanodbc; cd ..
-#   - ls -lah . ./contrib
+before_script:
+  # nanodbc for more tests
+  - cd ./contrib && git clone --recursive https://github.com/nanodbc/nanodbc.git && cd ..
+  - ls -lah . ./contrib
 
 script:
   - mkdir -p build
   - cd build
-  - cmake .. -G Ninja -DCMAKE_CXX_COMPILER=`which $CXX` -DCMAKE_C_COMPILER=`which $CC` $CMAKE_FLAGS -DTEST_DSN=clickhouse_localhost -DTEST_DSN_W=clickhouse_localhost_w && cmake --build . --target all
+  # MS ODBC uses UCS2 for widechars (2 bytes per char)  
+  # unixODBC do the same by default (2 bytes per char), but can be switched to wchar_t (4 bytes) (SQL_WCHART_CONVERT)
+  # iODBC always use wchar_t (4 bytes)
+  # so driver compiled for iODBC will not work properly with unixODBC software
+  - |
+    if [[ "$ODBC_LIB" =~ "iodbc" ]]; then
+      CMAKE_FLAGS="-DFIND_IODBC_FIRST=1 $CMAKE_FLAGS"
+    else
+      CMAKE_FLAGS="-DFIND_UNIXODBC_FIRST=1 $CMAKE_FLAGS"
+    fi
+  - cmake -G Ninja -DCMAKE_CXX_COMPILER=`which $CXX` -DCMAKE_C_COMPILER=`which $CC` -DTEST_DSN=clickhouse_localhost -DTEST_DSN_W=clickhouse_localhost_w $CMAKE_FLAGS ..
+  - cmake --build . --target all
   # unit-tests
   - ctest -R '.*-ut.*'
   # more complicated tests
-  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then echo "Unable to test on OSX since docker is not supported on Travis OSX agents."; exit ${TRAVIS_TEST_RESULT}; fi
-  # Setting up test prerequisites: docker, pyodbc, odbc module for perl, isql (or iusql or other)
-  - sudo apt install -y perl libdbi-perl libdbd-odbc-perl python-pyodbc unixodbc unixodbc-dev
-  # Using absolute path to compiler to fix setup.py issue of pyodbc: unable to execute 'clang': No such file or directory
-  #- sudo CC=`which $CC` CXX=`which $CXX` pip install pyodbc -V --disable-pip-version-check
-  # To simplify things a bit, start ClickHouse server in docker
-  - echo "Clickhouse image ${CLICKHOUSE_SERVER_IMAGE}";
-  - docker pull ${CLICKHOUSE_SERVER_IMAGE}
-  - CLICKHOUSE_SERVER_CONTAINER=$(docker run  -d ${CLICKHOUSE_SERVER_IMAGE})
-  - CLICKHOUSE_SERVER_IP=$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' ${CLICKHOUSE_SERVER_CONTAINER})
-  - docker ps -a
-  - docker stats -a --no-stream
+  - |
+    if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+      # install more deps for python & perl to run more tests    
+      sudo PERL_MM_USE_DEFAULT=1 cpan -T DBD::ODBC
+      curl https://bootstrap.pypa.io/get-pip.py | sudo python
+      # it looks like that older version works with iodbc out of the box on Mac
+      # https://github.com/mkleehammer/pyodbc/commit/aa178bfd25afd6fb90c77d054c1bcf0a6aa54411
+      sudo pip install pyodbc==3.0.7 
+      CLICKHOUSE_SERVER_IP='localhost'
+      ODBC_DRIVER_PATH=$(realpath driver/libclickhouseodbc.dylib)
+      ODBC_DRIVERW_PATH=$(realpath driver/libclickhouseodbcw.dylib)
+      # echo "Unable to test on OSX since docker is not supported on Travis OSX agents."; exit ${TRAVIS_TEST_RESULT};
```

Comment:
@filimonov is this commented out line of code still worth keeping?

Reply:
we can remove that. Can you try to run Python & perl tests on mac?

Reply:
Currnlty all needed conponents are installed and should work, but it looks like disabled on ctest side

Reply:
I removed that one, but for the all other commented out sections - better to address them as a separate PR of actualizing the whole travis test matrix.

Reply:
@filimonov I've enabled `ctest`'s tests in macOS too, and added some heuristics for detecting the actual ODBC Driver Manager used by Python's pyodbc and Perl's DBI::ODBC, and `ctest`'s tests will be enabled on each platform and environment according to that (following the rule: driver must be built against and use the same driver manager as the client.)

Reply:
Perfect! For 2 failing build - can we also check if driver compiled against iodbc can't work with unixODBC driver manager? 

currently we install iodbc to build driver, and after that we were installing unixODBC *extra*. What if we will try to install that *instead*? I.e. install iodbc + libs &headers only at build time, and afterwards remove it and install unixODBC driver manager instead? 

I was looking on mysql odbc driver and it looks like it's portable between iodbc and unixodbc.

Reply:
No, iODBC compiled driver/client can't interoperate in general with UnixODBC compiled one.

As far as static vs dynamic linking is concerned, static linking is worse since it doesn;t enforce the correct usage.

> I.e. install iodbc + libs &headers only at build time
I am revisiting travis config, hopefully those shaky things will be gone.